### ISSUE TITLE
[Q]Handling Asynchronous Step Logging for Actor and Critic Networks in Reinforcement Learning with wandb.log

### ISSUE THREAD

**Issue by fecet** [wandb/wandb#5734]
### Issue description

Hello wandb developers,

I've encountered a challenge while using wandb to record logs of two distinct neural networks during reinforcement learning. In my scenario, I'm training an actor and a critic separately, and each of them is associated with `actor/loss` and `critic/loss` logs respectively.

Because of the sequential training order, early stopping, training warmup and other factors, the global_samples (the `step` argument in wandb.log) for the actor and critic are not in sync. That is, the actor's `step` is often ahead of the critic's.

As a result, there are times when I need to log to a "historical" step. For the same wandb run, my behavior appears to be writing some historical logs, but these are due to the different steps between the two networks.

However, wandb.log seems to show some undefined behavior when handling this kind of situation, such as only logging the result of a certain step and ignoring the others.

### Question

Are there any suggestions or methods you could recommend to help me accurately log training steps under these conditions?

### Environment
- Wandb version: latest
- OS: Ubuntu 22.04
- Python version: 3.9

Any advice would be greatly appreciated. Thanks in advance!

**Comment by fecet** [comment]
Great! Thanks
